File: BOJ/완전탐색/7568/7568.cpp
// 7568 덩치
// https://www.acmicpc.net/problem/7568
#include <iostream>
#include <vector>
// 실5
using namespace std;

void solution(int N, vector<pair<int,int>> info) {
    vector<int> answer;

    for (int i = 0; i < info.size(); ++i)
    {
        pair<int,int> cur = info[i];
        int cnt = 1;
        for (int j = 0; j < info.size(); ++j)
        {
            if (i == j) continue;

            if (cur.first < info[j].first && cur.second < info[j].second)
            {
                cnt++;
            }
        }
        answer.push_back(cnt);
    }

    for (int a: answer)
    {
        cout << a << ' ';
    }
    cout << '\n';
    return;
}

int main(void) {
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    int N;
    vector<pair<int,int>> info;

    cin >> N;
    info = vector<pair<int,int>> (N);
    for (int i = 0; i< N; ++i)
    {
        cin >> info[i].first >> info[i].second;
    }

    solution(N, info);

    return 0;
}

File: BOJ/완전탐색/7568/7568.py
# 7568 덩치
# https://www.acmicpc.net/problem/7568
import sys
input = lambda : sys.stdin.readline().rstrip()

N = int(input())
arr = []
for _ in range(N):
    arr.append(list(map(int,input().split())))
answer = []

for i in range(N):
    rank = 1
    for j in range(N):
        if i == j: # 자기 자신
            continue
        # 확실히 크다면
        if arr[i][0] < arr[j][0] and arr[i][1] < arr[j][1]:
            rank += 1

    answer.append(rank)

print(*answer)